### DISCUSSION TITLE
Video Doorbell POE Firmware and Reolink Desktop App Update Check (API Request)

### DISCUSSION THREAD

**Discussion by pixeldoc2000** [AT0myks/reolink-fw-archive#40] [Firmware sharing]
I have sniffed the API Request the Reolink Client (Desktop) does for Update Check of a device:

# HTTPS Request to API
```
https://apis.reolink.com/v1.0/devices/roms/latests/?build=v3.0.0.2033_23041302&uid=xxxxxxxxxxxxxxxxxx&method=manual&mode=full&language=en-en
```
replace uid with your UID or 0 for example

# JSON Answer
```
{"needupdate":1,"board":"","version":"v3.0.0.2033_23090402","changelog":"","files":[{"type":"full","url":"https://firmwares.cdn.reolink.com/firmwares/Reolink-Video-Doorbell-PoE.2033_23090402.DB_566128M5MP_P.paks","name":"Reolink-Video-Doorbell-PoE.2033_23090402.DB_566128M5MP_P.paks","crc":-1815409288,"size":37119117,"ver":"v3.0.0.2033_23090402"}]}
```
This will give you the current / next Firmware for your device / firmware.

These Versions are only available via Update Check with Desktop App, not available on their Website:
v3.0.0.2555_23080702 (Beta ?):
https://firmwares.cdn.reolink.com/firmwares/Reolink-Video-Doorbell-PoE.2555_23080702.DB_566128M5MP_P.paks 
v3.0.0.2033_23090402 (this one is missing in the list):
https://firmwares.cdn.reolink.com/firmwares/Reolink-Video-Doorbell-PoE.2033_23090402.DB_566128M5MP_P.paks

Reolink Desktop App will update POE Device in this order: v3.0.0.2033_23041302 -> v3.0.0.2033_23090402 -> v3.0.0.2555_23080702

Sadly there is no Changelog.

**Comment by AT0myks** [comment]
That's very interesting, thank you. Some time ago I tried finding the endpoint that a few newer cameras use to auto update but I didn't search hard enough to find something. I did not think about trying with the desktop app. Two things that I noticed:
- looks like the `uid` parameter can be anything other than `0`, and it must be present
- if you remove the `mode` parameter the result is XML instead of JSON

Do you know what other values are valid for `method` and `mode`? It's probably not important but just in case.
A quick script testing all the versions that are in the archive right now only found 4 results that aren't new firmwares. But when I tried with the version from @ryan-roberts's RLC-510A in #39 it did return a [new firmware](https://firmwares.cdn.reolink.com/firmwares/RLC-510A.2839_23102300.IPC_MS1NA45MP.paks).

What's even more interesting is that these `paks` files have a different format from the `pak` ones. Seeing the `s` I thought they might be encrypted or something but fortunately it doesn't seem to be the case. Their magic is `SKAP` (`PAKS`) and in the two files you link here the string `SKAPLOER` (`REOL`, probably for Reolink) appears quite a few times (or maybe it's simply a coincidence), same for  `DIDB`. `SGMI` appears before the name of each section. For now I'm not going to add these links, I'd like to understand this new format first.

**Comment by pixeldoc2000** [comment]
> Do you know what other values are valid for method and mode? It's probably not important but just in case.

Sadly, no. I think one would have to RE the Desktop App or capture more "Update" Traffic, to see if there is more.

Yes, noticed the different format too. Did not try to apply the `.paks` Firmware manually to my Doorbell yet. If I find the time, I will sniff the Update Process via Desktop App.

Maybe `.paks` files can only be applied via API to the camera?

I've used this method on Windows to sniff the TLS / SSL Traffic between the Reolink Desktop App and the Device: https://unit42.paloaltonetworks.com/wireshark-tutorial-decrypting-https-traffic/

**Comment by pixeldoc2000** [comment]
Latest Firmware on Reolink Support Website: v3.0.0.2033_23041302 (https://support.reolink.com/c/video-doorbell-poe/?search=doorbell%20poe#274312295278239-download-f)

Latest Firmware via Reolink Client around December 2023: v3.0.0.2033_23090402

Current Reolink Client 8.15.6 does still query the same API URL.

API Query for all "known" Firmware Version:
```
https://apis.reolink.com/v1.0/devices/roms/latests/?build=v3.0.0.2033_23090402&uid=xxxxxxxxxxxxxxxx&method=manual&mode=full&language=en-en##
{"board":"","needupdate":0,"changelog":"","version":""}

https://apis.reolink.com/v1.0/devices/roms/latests/?build=v3.0.0.2033_23041302&uid=xxxxxxxxxxxxxxxx&method=manual&mode=full&language=en-en##
{"board":"","needupdate":0,"changelog":"","version":""}

https://apis.reolink.com/v1.0/devices/roms/latests/?build=v3.0.0.2555_23080702&uid=xxxxxxxxxxxxxxxx&method=manual&mode=full&language=en-en##
{"board":"","needupdate":0,"changelog":"","version":""}

https://apis.reolink.com/v1.0/devices/roms/latests/?build=v3.0.0.2017_23041202&uid=xxxxxxxxxxxxxxxx&method=manual&mode=full&language=en-en##
{"board":"","needupdate":0,"changelog":"","version":""}
```

Look like there is no Firmware Update available via API / Reolink Client at all ATM for this device?
